source/browseMode.py

```diff
@@ -2001,6 +2041,129 @@ def _iterNotLinkBlock(self, direction="next", pos=None):
 				yield TextInfoQuickNavItem("notLinkBlock", self, textRange)
 			item1=item2
 
+	STYLE_ATTRIBUTES = frozenset([
+		'background-color',
+		'color',
+		'font-family',
+		'font-size',
+		'bold',
+		'italic',
+		'marked',
+		'strikethrough',
+		'text-line-through-style',
+		'underline',
+		'text-underline-style',
+	])
+
+	def _extractStyles(
+			self,
+			info: textInfos.TextInfo,
+	) -> "textInfos.TextInfo.TextWithFieldsT":
+		"""
+		This function calls TextInfo.getTextWithFields(), and then processes fields in the following way:
+		1. Highlighted(marked) text is currently reported as Role.MARKED_CONTENT, and not formatChange.
+		For ease of further handling we create a new boolean format field "marked"
+		and set its value according to presence of Role.MARKED_CONTENT.
+		2. Then we drop all control fields, leaving only formatChange fields and text.
+		"""
+		stack = [{}]
+		result = []

```

Comment:
can these get type info added. It's useful for objects like to know what child item types to expect